homeassistant/components/bosch_shc/sensor.py

```diff
@@ -0,0 +1,401 @@
+        return f"{self._device.serial}_airquality"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Air Quality"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.combined_rating.name
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            "rating_description": self._device.description,
+            "temperature_rating": self._device.temperature_rating.name,
+            "humidity_rating": self._device.humidity_rating.name,
+            "purity_rating": self._device.purity_rating.name,
+        }
+
+
+class PowerSensor(SHCEntity):
+    """Representation of a SHC power reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_power"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Power"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.powerconsumption
+
+    @property
+    def device_class(self):
+        """Return the class of this device."""
+        return DEVICE_CLASS_POWER
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the sensor."""
+        return POWER_WATT
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return "mdi:flash"
+
+
+class EnergySensor(SHCEntity):
+    """Representation of a SHC energy reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_energy"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Energy"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.energyconsumption / 1000.0
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the sensor."""
+        return ENERGY_KILO_WATT_HOUR
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return "mdi:gauge"
+
+
+class BatterySensor(SHCEntity):
+    """Representation of a SHC battery reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_battery"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Battery"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if (
+            self._device.batterylevel
+            == SHCBatteryDevice.BatteryLevelService.State.CRITICAL_LOW
```

Comment:
We also provide binary sensor device class for battery which may be applicable instead.

Reply:
I will move this to the binary sensor platform.